src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -125,6 +125,77 @@ private static void SendToRecvFromAsync_Datagram_UDP(IPAddress leftAddress, IPAd
             }
         }
 
+        private static void SendToRecvFromAsync_UdpClient_Datagram_UDP(IPAddress leftAddress, IPAddress rightAddress)
+        {
+            const int DatagramSize = 256;
+            const int DatagramsToSend = 256;
+            const int AckTimeout = 1000;
+            const int TestTimeout = 30000;
+
+            var left = new UdpClient(new IPEndPoint(leftAddress, 0));
+            var right = new UdpClient(new IPEndPoint(rightAddress, 0));
+
+            var leftEndpoint = (IPEndPoint)left.Client.LocalEndPoint;
+            var rightEndpoint = (IPEndPoint)right.Client.LocalEndPoint;
+
+            var receiverAck = new ManualResetEventSlim();
+            var senderAck = new ManualResetEventSlim();
+
+            var receivedChecksums = new uint?[DatagramsToSend];
+            int receivedDatagrams = 0;
+
+            var receiverTask = Task.Run(async () =>
+            {
```

Comment:
Should you use `var` here or use `Task` ?  Comment applies to other instances of this in the file.